clients/admin-ui/src/features/datastore-connections/ConnectionGridItem.tsx

```diff
@@ -76,6 +51,51 @@ const TestData: React.FC<TestDataProps> = ({ succeeded, timestamp }) => {
   );
 };
 
+const useConnectionGridItem = () => {
+  const getConnectorDisplayName = (connectionType: ConnectionType): string => {
+    if (Object.values(ConnectionType).includes(connectionType)) {
+      return `${capitalize(connectionType)} Database Connector`;
+    }
+
+    let value: string;
+    switch (connectionType) {
+      case ConnectionType.HTTPS:
+        value = "HTTPS Connector";
+        break;
+      case ConnectionType.MANUAL:
+        value = "Manual Connector";
+        break;
+      case ConnectionType.SAAS:
+        value = "Saas Connector";
+        break;
+      default:
+        value = "Unknown Connector";
+        break;
+    }
+
+    return value;
+  };
+
+  const getImageSrc = (data: DatastoreConnection): string => {
+    const item = [...ConnectionTypeLogoMap].find(
+      ([k]) =>
+        (data.connection_type.toString() !== ConnectionType.SAAS &&
+          data.connection_type.toString() === k) ||
+        (data.connection_type.toString() === ConnectionType.SAAS &&
+          data.saas_config?.type?.toString() === k.toString())
+    );
+    const path = item
```

Comment:
Can we jut return directly here instead of introducing a new var?

```suggestion
    return item
        ? CONNECTOR_LOGOS_PATH + item[1]
        : FALLBACK_CONNECTOR_LOGOS_PATH;
```